flink-runtime/src/test/java/org/apache/flink/runtime/net/SSLUtilsTest.java

```diff
@@ -52,6 +60,23 @@
 	private static final String KEY_STORE_PASSWORD = "password";
 	private static final String KEY_PASSWORD = "password";
 
+	public static final List<String> AVAILABLE_SSL_PROVIDERS;
+	static {
+		if (OpenSsl.isAvailable()) {
+			AVAILABLE_SSL_PROVIDERS = Arrays.asList("JDK", "OPENSSL");
```

Comment:
Do we test for this somehow on travis? Manually? Is there some kind of instruction how to make `OpenSsl` available during testing?

Reply:
For now, we'd have to have a custom flink-shaded build around*. There are licensing issues that prevent us from going the easy way and the actual way of providing openSSL in a safe way for the Flink distribution is not set in stone yet. I'll create a follow-up PR with proper documentation once this has been defined.

* I have two WIP branches for this: https://github.com/NicoK/flink-shaded/commits/flink-9816.static-6.0 and https://github.com/NicoK/flink-shaded/tree/flink-9816

Reply:
I enriched this PR with a commit that enables the custom flink-shaded builds with static openSSL builds for Travis, just to see whether the tests run through. We would have to drop that commit during the merge though.